### DISCUSSION TITLE
Basemap EPSG missing when using pyinstaller for creating an executable

### DISCUSSION THREAD

**Discussion by ArslaanMB** [pyinstaller/pyinstaller#7824] [PyInstaller] [answered]
Hi,
For an application, I am trying to generate an executable where basemap library is being used. I imported the library using "from mpl_toolkits.basemap import Basemap,basemap_data" in the .py code

and for the .spec file I am using the
additional_packages = ['mpl_toolkits.basemap', 'mpl_toolkits.basemap_data', 'mpl_toolkits.basemap_data.epsg']

and connecting it to hidden import through: hiddenimports=additional_packages,

but the generated exe is resulting into EPSG missing error.

Can anyone please help me? I tried the method here but it didnot work: https://stackoverflow.com/questions/17749296/get-pyinstaller-to-import-basemap

Below is my full .spec file:

-- mode: python ; coding: utf-8 --
import sys ; sys.setrecursionlimit(sys.getrecursionlimit() * 5)
import mpl_toolkits.basemap
import os

additional_packages = ['mpl_toolkits.basemap', 'mpl_toolkits.basemap_data', 'mpl_toolkits.basemap_data.epsg']

block_cipher = None

a = Analysis(
['QC_checks_on_USACE_data_portable_CHS_regions.py'],
pathex=[],
binaries=[],
datas=[],
hiddenimports=additional_packages,
hookspath=[],
hooksconfig={},
runtime_hooks=[],
excludes=[],
win_no_prefer_redirects=False,
win_private_assemblies=False,
cipher=block_cipher,
noarchive=False,
)
pyz = PYZ(a.pure, a.zipped_data, cipher=block_cipher)

exe = EXE(
pyz,
a.scripts,
a.binaries,
a.zipfiles,
a.datas ,
[],
name='QC_checks_on_USACE_data_portable_CHS_regions_v3',
debug=False,
bootloader_ignore_signals=False,
strip=False,
upx=True,
upx_exclude=[],
runtime_tmpdir=None,
console=False,
disable_windowed_traceback=False,
argv_emulation=False,
target_arch=None,
codesign_identity=None,
entitlements_file=None,
)

Please see the error below
![image](https://github.com/pyinstaller/pyinstaller/assets/106843025/bf591b03-3152-4417-a1a5-336cae74ec58)

**Comment by rokm** [comment, marked as answer]
You need to collect data files from `mpl_toolkits` package.

Either generate the new spec file with `--collect-data mpl_toolkits` added to the PyInstaller command, or modify the existing spec:

```
from PyInstaller.utils.hooks import collect_data_files

[...]

datas = []
datas += collect_data_files('mpl_toolkits')

[...]

a = Analysis(
    ...
    datas=datas,
    ...
)
